ledger/src/snapshot_utils.rs

```diff
@@ -37,14 +37,20 @@ pub const TAR_VERSION_FILE: &str = "version";
 const MAX_SNAPSHOT_DATA_FILE_SIZE: u64 = 32 * 1024 * 1024 * 1024; // 32 GiB
 const VERSION_STRING_V1_1_0: &str = "1.1.0";
 const VERSION_STRING_V1_2_0: &str = "1.2.0";
-const OUTPUT_SNAPSHOT_VERSION: SnapshotVersion = SnapshotVersion::V1_2_0;
+const DEFAULT_SNAPSHOT_VERSION: SnapshotVersion = SnapshotVersion::V1_2_0;
 
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub enum SnapshotVersion {
```

Comment:
Because this is now exposed to the cli, could you make this more user friendly? `impl FromStr for SnapshotVersion`:
https://github.com/solana-labs/solana/pull/10536/files#diff-0f0af8fcfa1444e1b07d671c5bfa4a69R70

- `fmt!` to embed actual bad version string user supplied: "unsupported snapshot version"
- Allow `v` prefixing like `v1.2.0`. Precedent: https://github.com/solana-labs/solana/pull/8297

Reply:
Caller to str::parse supplies the input string and thus has it, no need for Error object to allocate heap memory & copy input.

For CLI, input validator constructs user friendly error message flagging supplied string for user.

FromStr now strips off /^[vV]/; note this necessarily also applies to the contents of the version file in the snapshot archive. Snapshot versions will not change with every minor release, having them be so similar is possibly misleading.

In any event the purpose of this PR is not to define the snapshot version identifier but merely to expose it (whatever it is) to the command line for use as an optional advanced / debug option.